src/driver/DriverFactory.ts

```diff
@@ -27,48 +8,59 @@ export class DriverFactory {
     /**
      * Creates a new driver depend on a given connection's driver type.
      */
-    create(connection: DataSource): Driver {
-        const { type } = connection.options
+    create(dataSource: DataSource): Driver {
+        const { type } = dataSource.options
+        return new (this.getDriver(type))(dataSource)
+    }
+
+    private getDriver(type: DataSource["options"]["type"]): DriverConstructor {
```

Comment:
```suggestion
    private async getDriver(type: DataSource["options"]["type"]): DriverConstructor {
```

Reply:
this would require making `create` async as well. and that could break the public api.

Reply:
Ah, now I see. https://nodejs.org/docs/latest-v15.x/api/module.html#module_module_createrequire_filename this could be also used, maybe. Just to not be stuck with some commonJS syntax.
But otherwise you are absolutly correct.